storage/mdadm/include/include.sh

```diff
@@ -0,0 +1,619 @@
+#!/bin/bash
+
+# Copyright (c) 2016 Red Hat, Inc. All rights reserved.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Author: Chnaghui Zhong <czhong@redhat.com>
+
+rpm -q mdadm || yum install -y mdadm
+JOURNAL_SUPPORT=0
+info=`mdadm --create --help | grep -o "write-journal"`
+if [ "$info" = "write-journal" ]; then
+        JOURNAL_SUPPORT=1
+fi
+
+#Install fio from upstream
+function install_fio() {
+
+	git_url=git://git.kernel.org/pub/scm/linux/kernel/git/axboe/fio.git
+
+	yum install libaio-devel zlib-devel -y
+	git clone $git_url
+	echo "INFO: Installing Fio"
+	cd fio  &&./configure && make && make install
+	which fio
+	if [ $? -ne 0 ]; then
+		echo "FAIL: Fio not succesffully installed"
+		exit 1
+	fi
+
+	echo "INFO: Fio succesfully installed"
+}
+
+#Install dt
+function install_dt () { 
+
+	wget http://www.scsifaq.org/RMiller_Tools/ftp/dt/dt-source.tar.gz
+	tar xvf dt-source.tar.gz
+	cd dt.d-WIP/
+	cp -p Makefile.linux Makefile
+	make
+	cp dt /usr/bin
+}
+
+#----------------------------------------------------------------------------#
+# MD_Create_RAID ()
+# Usage:
+#   Create md raid.
+# Parameter:
+# 	$level				# like 0, 1, 3, 5, 10, 50                       
+# 	$dev_list			# like 'sda sdb sdc sdd'
+# 	$raid_dev_num		# like 3                       
+# 	$spar_dev_num		# like 2                                    
+# 	$chunk				# like 64                                     
+# Returns:
+#   Return code:
+#       0 on success
+#       1 if something went wrong.
+#   Return string:
+#       RETURN_STR		# $md_raid like '/dev/md0'
+#		MD_DEVS			# $raid_dev_list like '/dev/sda /dev/sdb'
+#----------------------------------------------------------------------------#
+
+function MD_Create_RAID (){
+	EX_USAGE=64 # Bad arg format
+	if [ $# -lt 3 ]; then
+		echo 'Usage: MD_Create_RAID $level $dev_list $raid_dev_num \
+		[$spar_dev_num] [$chunk]'
+		exit "${EX_USAGE}"
```

Comment:
This looks like a test / beaker XML configuration problem, rather then a real kernel failure. So I have the same question as above -- we don't want to report this to developers.

Reply:
If the instruction fail to execute, just use "exit", or don't use "exit" to return its own information and exit code. Which method is appropriate?

Reply:
Depends on how the result is interpreted. This file was removed but if you check the exit code in the test then you can exit with code.
When checking the exit code, the rule is: fail if it's a kernel bug, abort if it is not.

Does this answer your question? I think the current code looks ok